src/Covenant/Analysis/Poetry/PoetryLicenseParser.cs

```diff
@@ -0,0 +1,120 @@
+using System.Text.RegularExpressions;
+
+namespace Covenant.Analysis.Poetry;
+
+internal static class PoetryLicenseParser
+{
+    public static BomLicense? Parse(string sitePackagesPath, PoetryLockPackage package, AnalysisContext context)
+    {
+        var packageName = package.Name!.Replace("-", "_");
+        var jsonMetadataPath = System.IO.Path.Join(sitePackagesPath, $"{packageName}-{package.Version}.dist-info", "metadata.json");
+        var metadataPath = System.IO.Path.Join(sitePackagesPath, $"{packageName}-{package.Version}.dist-info", "METADATA");
+        var licensePath = System.IO.Path.Join(sitePackagesPath, $"{packageName}-{package.Version}.dist-info", "LICENSE");
+
+        List<(string Path, Func<string, string?> Parser)> metadataTypes = [
+            (jsonMetadataPath, ParseJsonMetadataFile),
+            (metadataPath, ParseMetadataFile),
+            (licensePath, ParseLicenseFile)
+        ];
+
+        string? licenseString = null;
+
+        foreach (var m in metadataTypes)
+        {
+            if (!File.Exists(m.Path))
+            {
+                continue;
+            }
+
+            licenseString = m.Parser(m.Path);
+
+            if (licenseString != null)
+            {
+                break;
+            }
+        }
+
+        if (licenseString == null)
```

Comment:
Very minor style nit, but elsewhere in this PR you've used the new `is not null` syntax, so this is slightly inconsistent.

There's a somewhat obscure reason for preferring the new syntax: when types overload the `==` operator, writing `== null` might not actually do what you expect. The newer `is null` syntax unambiguously checks whether the reference is `null`, regardless of any operator overloads it might have defined.